catroid/src/main/java/org/catrobat/catroid/stage/VisualPlacementActivity.java

```diff
@@ -0,0 +1,304 @@
+		} else if (currentSprite.getLookList().size() != 0) {
+			objectLookPath = currentSprite.getLookList().get(0).getFile().getAbsolutePath();
+			spriteBitmap = BitmapFactory.decodeFile(objectLookPath, options);
+		} else {
+			Log.e(TAG, "this sprite has no Looks");
+			spriteBitmap = BitmapFactory.decodeResource(getResources(), R.drawable.ic_launcher);
+		}
+
+		int scaledSpriteBitmapWidth = (int) (spriteBitmap.getWidth() * scaleWidthRatio);
+		int scaledSpriteBitmapHeight = (int) (spriteBitmap.getHeight() * scaleHeightRatio);
+
+		Bitmap scaledSpriteBitmap = Bitmap.createScaledBitmap(spriteBitmap, scaledSpriteBitmapWidth, scaledSpriteBitmapHeight, false);
+		spriteBitmap.recycle();
+		imageView = new ImageView(this);
+		imageView.setImageBitmap(scaledSpriteBitmap);
+		imageView.setScaleType(ImageView.ScaleType.CENTER);
+
+		if (scaleX > 0.01) {
+			imageView.setScaleX(scaleX);
+		}
+
+		if (scaleY > 0.01) {
+			imageView.setScaleY(scaleY);
+		}
+
+		if (rotation != 0) {
+			imageView.setRotation(rotation - 90);
+		}
+
+		frameLayout.addView(imageView);
+		toolbar.bringToFront();
+		frameLayout.setOnTouchListener(this);
+	}
+
+	@Override
+	public boolean onTouch(View view, MotionEvent event) {
+		float currentX = event.getRawX();
+		float currentY = event.getRawY();
+		switch (event.getAction()) {
+			case MotionEvent.ACTION_DOWN:
+				previousX = currentX;
+				previousY = currentY;
+				startX = currentX;
+				startY = currentY;
+				break;
+			case MotionEvent.ACTION_CANCEL:
+			case MotionEvent.ACTION_UP:
+				if (Math.abs(currentX - startX) < 5 && Math.abs(currentY - startY) < 5) {
+					imageView.animate()
+							.translationX(event.getX() - (float) view.getWidth() / 2)
+							.translationY(event.getY() - (float) view.getHeight() / 2)
+							.setDuration(0)
+							.start();
+				}
+				break;
+			case MotionEvent.ACTION_MOVE:
+				float dX = currentX - previousX;
+				float dY = currentY - previousY;
+				imageView.setX(imageView.getX() + dX);
+				imageView.setY(imageView.getY() + dY);
+				previousX = currentX;
+				previousY = currentY;
+				break;
+		}
+		xCoord = imageView.getX();
+		yCoord = -imageView.getY();
+		coordinatesChanged = true;
+		return true;
+	}
+
+	@Override
+	protected void onStart() {
+		super.onStart();
+		getWindow().setFlags(WindowManager.LayoutParams.FLAG_FULLSCREEN, WindowManager.LayoutParams.FLAG_FULLSCREEN);
+		getWindow().addFlags(WindowManager.LayoutParams.FLAG_TRANSLUCENT_STATUS);
+	}
+
+	@Override
+	public void onBackPressed() {
+		if (coordinatesChanged) {
+			showSaveChangesDialog(this);
+		} else {
+			super.onBackPressed();
+		}
+	}
+
+	@Override
+	public void onClick(DialogInterface dialog, int which) {
+		switch (which) {
+			case BUTTON_POSITIVE:
+				saveCoordinates();
+				break;
+			case BUTTON_NEGATIVE:
+				ToastUtil.showError(this, R.string.formula_editor_changes_discarded);
+				finish();
+				break;
+		}
+	}
+
+	private void saveCoordinates() {
```

Comment:
please rename to finishWithResult or something similar. since its not saving anything, and also finishing the activity